sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvoker.java

```diff
@@ -43,7 +46,23 @@
    *
    * @param c The {@link DoFn.ProcessContext} to invoke the fn with.
    * @param extra Factory for producing extra parameter objects (such as window), if necessary.
+   * @return The {@link DoFn.ProcessContinuation} returned by the underlying method,
+   *   or {@link DoFn.ProcessContinuation#stop()} if it returns {@code void}.
    */
-  void invokeProcessElement(
+  DoFn.ProcessContinuation invokeProcessElement(
       DoFn<InputT, OutputT>.ProcessContext c, DoFn.ExtraContextFactory<InputT, OutputT> extra);
+
+  /** Invoke the {@link DoFn.GetInitialRestriction} method on the bound {@link DoFn}. */
+  <RestrictionT> RestrictionT invokeGetInitialRestriction(InputT element);
+
+  /** Invoke the {@link DoFn.GetRestrictionCoder} method on the bound {@link DoFn}. */
+  <RestrictionT> Coder<RestrictionT> invokeGetRestrictionCoder();
+
+  /** Invoke the {@link DoFn.SplitRestriction} method on the bound {@link DoFn}. */
```

Comment:
Would probably be good to document these more. Eg., what should a runner pass as "numParts", etc.

Eg., write these as documentation for runners and such that will be invoking `DoFn` methods, and the Javadoc in `DoFn` is more for people who are writing DoFns.